addons/binding/org.openhab.binding.enocean/ESH-INF/thing/channels.xml

```diff
@@ -0,0 +1,268 @@
+	</channel-type>
+
+	<channel-type id="fanSpeedStage">
+		<item-type>String</item-type>
+		<label>Fan speed</label>
+		<state readOnly="true">
+			<options>
+				<option value="-1">Auto</option>
+				<option value="0">Stage 0</option>
+				<option value="1">Stage 1</option>
+				<option value="2">Stage 2</option>
+				<option value="3">Stage 3</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="occupancy">
+		<item-type>Switch</item-type>
+		<label>Occupancy</label>
+		<description>Occupancy button pressed or released.</description>
+		<category>Switch</category>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="contact">
+		<item-type>Contact</item-type>
+		<label>Contact</label>
+		<description>Contact channel</description>
+	</channel-type>
+
+	<channel-type id="windowHandleState">
+		<item-type>String</item-type>
+		<label>Window handle state</label>
+		<description>Describes the window handle state</description>
+		<state readOnly="true">
+			<options>
+				<option value="OPEN">Open</option>
+				<option value="TILTED">Tilted</option>
+				<option value="CLOSED">Closed</option>
+			</options>
+		</state>
+	</channel-type>
+
+	<channel-type id="instantpower">
+		<item-type>Number:Power</item-type>
+		<label>Instant Power</label>
+		<description>Instant power consumption in Watts</description>
+		<state pattern="%d %unit%" readOnly="true" />
+	</channel-type>
+
+	<channel-type id="totalusage">
+		<item-type>Number:Energy</item-type>
+		<label>Total Usage</label>
+		<description>Used energy in Kilowatt hours</description>
+		<state pattern="%d %unit%" readOnly="true" />
+	</channel-type>
+
+	<!-- Teach in Channel -->
+	<channel-type id="teachInCMD" advanced="true">
+		<item-type>Switch</item-type>
+		<label>Teach In</label>
+		<description>The teach in channel allows to send a teach in msg.</description>
+		<category>Switch</category>
+		<config-description>
+			<!-- <parameter name="teachInType" type="text"> <label>Teach in type</label> <description>Type of 4BS teach in message</description> 
+				<options> <option value="uniWoEEP">unidirectional without EEP</option> <option value="uniWEEP">unidirectional with EEP</option> 
+				<option value="biWEEP">bidirectional with EEP</option> </options> <limitToOptions>true</limitToOptions> </parameter> -->
+			<parameter name="manufacturerId" type="text">
+				<label>Manufacturer Id</label>
+				<description>Id is used for 4BS teach in with EEP</description>
+			</parameter>
+			<parameter name="teachInMSG" type="text">
+				<label>Teach in message</label>
+				<description>Use this message if teach in type and/or manufacturer id are unknown</description>
+			</parameter>
+		</config-description>
+	</channel-type>
+
+	<channel-type id="receivingState">
+		<item-type>String</item-type>
+		<label>Receiving status</label>
+		<state readOnly="true" />
+	</channel-type>
+
+	<!-- virtual rocker switch channel -->
+	<channel-type id="virtualRockerswitch">
```

Comment:
Doesn't this duplicate the trigger channel? In general no "virtual" channels should be provided by a binding.

Reply:
A VirtualRockerSwitch is used to send rocker switch messages (uses switch channels acts like a virtual rocker). A RockerSwitch receives messages (uses trigger channels, represents a physical rocker). Had a long discussion about seperating these concepts into two thingtypes to make it more simple. However I am open for any new suggestions.

Reply:
Hm, ok, I think I understand what it is technically doing, but let me try to come from the use case side: Having a virtual rocker by itself does not bring any value, it will just send telegrams that go to nirvana. So a user will actually always control something with it - so effectively, the state of the linked item reflects the state of the device that is switched (and not that of the virtualRocker, which actually does not have any state). So if we are switching other devices, my question is whether we cannot control those other devices through their own channels already (and if not, if such "control channels" could/should be added to them?

> Had a long discussion about seperating these concepts 

Is this somewhere available on Github? If so, please point me to it and I might ask less stupid questions in future ;-)

Reply:
This rocker switch simulation (VirtualRockerSwitch) is just meant for old EnOcean devices which do not support the more advanced and newer EEPs like CentralCommand. 

I really like the idea to hide this technical aspect by using the already existing things and configure them to use rocker switch simulation instead of their advanced EEPs. However as I defined the thing-types according to their EEP they use, I have to introduce a new thing-type *OldEnOceanDeviceControlledByRockerSwitchMessages*. This thing-type would have at least a switching, rollershutter and dimmer channel. Good point.
However to be still able to emit pure rocker switch messages I would like to keep the VirtualRockerSwitch and its switch profile as an unlisted thing-type. By the way are not the profiles intended for such a use case (do the transfer logic from items to things and vice versa)?

You can find a part of the discussion [here](https://github.com/fruggy83/openocean/issues/6).

Reply:
I changed the VirtualRockerSwitch thing label to Classic Device, as this thing is essentially intended for such old devices, which just support rocker switch messages. I did not change the channels and kept the virtualrockerswitch ones. However I added a new profile for linking these channels to a rollershutter item (reacts only to up/down command). In this way the advanced users can still use these channels in their rules (some users from the community are using these things already in this way) and normal user are able to simple link these channels to switch/rollershutter items. A dimmer profile can be discussed later ;).
ClassicDevices and RockerSwitches should be used together
RockerSwitch => system profile => Switch Item => switch profile => ClassicDevice